v1/v1util/zip.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1util
+
+import (
+	"bytes"
+	"compress/gzip"
+	"io"
+)
+
+// GzipReadCloser reads uncompressed input data from the io.ReadCloser and
+// returns an io.ReadCloser from which compressed data may be read.
+func GzipReadCloser(r io.ReadCloser) (io.ReadCloser, error) {
+	defer r.Close()
+
+	// TODO(mattmoor): How to avoid buffering this whole thing into memory?
```

Comment:
Does a Pipe help? It might add more complexity than you need. https://medium.com/stupid-gopher-tricks/streaming-data-in-go-without-buffering-3285ddd2a1e5

Reply:
I think it's ~exactly what I want, but I think I'll save that for a follow-up.